### ISSUE TITLE
no-extraneous-dependencies should not complain when importing file from project name

### ISSUE THREAD

**Issue by alexmngn** [benmosher/eslint-plugin-import#541]
Hi,
I have a React-native project named "MyProject".
When I import a file from `'MyProject/src/store'` (which is valid with react-native, which is the way to load something from the root of the project), it complains.
I think it should detect that's the name of the current project, which is valid, and shouldn't throw an error.

<img width="1233" alt="screen shot 2016-09-04 at 10 48 26" src="https://cloud.githubusercontent.com/assets/4203845/18228717/6363004c-728d-11e6-97df-4fdcc03d5108.png">

**Comment by wmertens** [comment]
I think this is a specific case of #479, see also #496.

**Comment by benmosher** [comment]
I think you may be able to solve this with the following resolver config:

```js
// MyProject/.eslintrc.json
{
  "settings": {
    "import/resolver": {
      "node": { "paths": [ ".." ] }
    }
  }
}
```
or, failing that:
```js
// MyProject/.eslintrc.js
exports.settings = {
  "import/resolver": {
    "node": { "paths": [ path.join(__dirname, "..") ] }
  }
}
```

Notably, I'm assuming your `.eslintrc.{js,json,yml}` is located in the `MyProject` folder. If not, adjust accordingly.

**Comment by benmosher** [comment]
If the first way works, let me know and I'll update [config/react-native.js](https://github.com/benmosher/eslint-plugin-import/blob/master/config/react-native.js).

Or if I can make the assumption that ESLint will be run from the project folder, could do `path.resolve("..")`, IIRC it resolves relative to CWD.

**Comment by benmosher** [comment]
@alexmngn et. al., check out #626: @guzart implemented a React Native resolver. that should fix this, I think.

**Comment by benmosher** [comment]
psst: added to import/order internal milestone to remember the react-native resolver when implementing the resolver-chooses-import-type API (#479)
